app/actions/tenant.go

```diff
@@ -202,3 +204,29 @@ func (action *UpdateTenantPrivacy) IsAuthorized(ctx context.Context, user *entit
 func (action *UpdateTenantPrivacy) Validate(ctx context.Context, user *entity.User) *validate.Result {
 	return validate.Success()
 }
+
+// UpdateTenantAllowingEmailAuth is the input model used to update tenant privacy settings
+type UpdateTenantAllowingEmailAuth struct {
+	IsAllowingEmailAuth bool `json:"isAllowingEmailAuth"`
+}
+
+// IsAuthorized returns true if current user is authorized to perform this action
+func (action *UpdateTenantAllowingEmailAuth) IsAuthorized(ctx context.Context, user *entity.User) bool {
+	return user != nil && user.Role == enum.RoleAdministrator
+}
+
+// Validate if current model is valid
+func (action *UpdateTenantAllowingEmailAuth) Validate(ctx context.Context, user *entity.User) *validate.Result {
+	result := validate.Success()
+
+	activeProviders := &query.ListActiveOAuthProviders{}
+	if err := bus.Dispatch(ctx, activeProviders); err != nil {
+		return validate.Failed("Cannot retrieve OAuth providers")
+	}
+
+	if len(activeProviders.Result) == 0 {
+		result.AddFieldFailure("isAllowingEmailAuth", "You cannot disable email authentication without any other provider enabled.")
```

Comment:
what about `You need to configure another authentication provider before disabling email authentication.`

Reply:
Yes of course, this is far better that my original sentence, my English level produce sometime really weird wordings :sweat_smile: 
Changed message